src/main/scala/com/fulcrumgenomics/bam/EstimatePoolingFractions.scala

```diff
@@ -0,0 +1,211 @@
+    |for the minor allele fractions at each SNP locus, using as inputs the individual
+    |sample's genotypes.  Only SNPs that are bi-allelic within the pooled samples are
+    |used.
+    |
+    |Various filtering parameters can be used to control which loci are used:
+    | --intervals will restrict analysis to variants within the described intervals
+    | --min-genotype-quality will filter out any site with any genotype with GQ < n
+    | --min-coverage-per-sample requires that the coverage of a site in the BAM be
+    |     >= min-coverage-per-sample * n_samples
+    | --min-mapping-quality filters out reads in the BAM with MQ < n
+    | --min-base-quality filters out bases in the BAM with Q < n
+  """)
+class EstimatePoolingFractions
+( @arg(flag="v", doc="VCF of individual sample genotypes.")  val vcf: PathToVcf,
+  @arg(flag="b", doc="Path to BAM file of sequencing data.") val bam: PathToBam,
+  @arg(flag="o", doc="Output file to write with pooling fractions.") val output: FilePath,
+  @arg(flag="l", minElements=0, doc="Zero or more set of regions to restrict analysis to.") val intervals: Seq[PathToIntervals] = Seq.empty,
+  @arg(flag="s", minElements=0, doc="Optional subset of samples from VCF to use.") val samples: Seq[String] = Seq.empty,
+  @arg(flag="g", doc="Minimum genotype quality. Use -1 to disable.") val minGenotypeQuality: Int = 30,
+  @arg(flag="c", doc="Minimum sequencing coverage per sample.") val minCoveragePerSample: Int = 12,
+  @arg(flag="m", doc="Minimum mapping quality.") val minMappingQuality: Int = 30,
+  @arg(flag="q", doc="Minimum base quality.") val minBaseQuality:Int = 30
+) extends FgBioTool with LazyLogging {
+  Io.assertReadable(vcf :: bam :: intervals.toList)
+
+  /* Class to hold information about a single locus. */
+  case class Locus(chrom: String, pos: Int, ref: Char, alt: Char, fractions: Array[Double], var observedFraction: Option[Double] = None)
+
+  override def execute(): Unit = {
+    val vcfReader = new VCFFileReader(vcf.toFile)
+    val sampleNames = pickSamplesToUse(vcfReader)
+    val intervals   = loadIntervals
+
+    // Get the expected fractions from the VCF
+    val vcfIterator = constructVcfIterator(vcfReader, intervals, sampleNames)
+    val loci = vcfIterator.map { v => Locus(
+      chrom = v.getContig,
+      pos = v.getStart,
+      ref = v.getReference.getBaseString.charAt(0),
+      alt = v.getAlternateAllele(0).getBaseString.charAt(0),
+      fractions = sampleNames.map { s => val gt = v.getGenotype(s); if (gt.isHomRef) 0 else if (gt.isHet) 0.5 else 1.0 }
+    )}.toArray
+
+    logger.info(s"Loaded ${loci.length} bi-allelic SNPs from VCF.")
+
+    // And fill in the observed fractions from the BAM
+    val locusIterator = constructBamIterator(loci)
+    locusIterator.zip(loci.iterator).foreach { case (locusInfo, locus) =>
+      if (locusInfo.getSequenceName != locus.chrom || locusInfo.getPosition != locus.pos) fail("VCF and BAM iterators out of sync.")
```

Comment:
Yes, entirely a sanity check.  I figure it's easy to check and if they went out of sync due to some kind of bug, whether here or down in htsjdk, that would be very bad.